packages/core/src/util/cache.js

```diff
@@ -1,16 +1,78 @@
+import { list, tableFromArrays, tableFromIPC, tableToIPC,  uint8, utf8 } from "@uwdata/flechette";
+
 const requestIdle = typeof requestIdleCallback !== 'undefined'
   ? requestIdleCallback
   : setTimeout;
 
+/**
+ * @typedef {import('@uwdata/flechette').Table | Promise<import('@uwdata/flechette').Table>} CacheEntry
+ * @typedef {Object} Cache
+ * @property {(key: string) => CacheEntry | undefined} get Retrieves a value from the cache.
+ * @property {(key: string, value: CacheEntry) => any} set Stores a value in the cache and returns it.
+ * @property {() => void} clear Clears all entries in the cache.
+ * @property {() => Uint8Array | null} export Exports the cache as an array of bytes in Arrow IPC binary format or null if the cache is empty.
```

Comment:
It looks like these methods export/import the entire cache in one go. I wonder if we instead want something more modular, that imports key/value pairs (string keys, Arrow IPC values) rather than a single Arrow IPC table containing all entries. Curious if you had an overriding reason for the current approach?

Reply:
One idea is to have the cache import/export explicitly work with key value pairs (e.g., `Map<string,Uint8Array>` or `[string,Uint8Array][]`). The publish module can then make its own decisions about how to bundle (or not bundle) these up for reading/writing cache content to external files. (p.s. note also that Arrow includes a [`map` type](https://idl.uw.edu/flechette/api/data-types#map) you could use if it helpfully simplifies later serialization.)